src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlQueueClient.cs

```diff
@@ -10,28 +10,22 @@
 using System.Threading;
 using System.Threading.Tasks;
 using EnsureThat;
+using MediatR;
 using Microsoft.Data.SqlClient;
 using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Core.Messages.Storage;
 using Microsoft.Health.Fhir.SqlServer.Features.Schema.Model;
 using Microsoft.Health.JobManagement;
-using Microsoft.Health.SqlServer.Features.Schema;
 using Microsoft.Health.SqlServer.Features.Storage;
 using JobStatus = Microsoft.Health.JobManagement.JobStatus;
 
 namespace Microsoft.Health.Fhir.SqlServer.Features.Storage
 {
-    public class SqlQueueClient : IQueueClient
+    public class SqlQueueClient : IQueueClient, INotificationHandler<StorageInitializedNotification>
```

Comment:
I think we should update the registration too for consistency with notification handlers.
https://github.com/microsoft/fhir-server/blob/8aea3059635f98acb1ca599253d67ff0a5b845e3/src/Microsoft.Health.Fhir.SqlServer/Registration/FhirServerBuilderSqlServerRegistrationExtensions.cs#L130-L133

to 

```
            services.RemoveServiceTypeExact<SqlQueueClient, INotificationHandler<StorageInitializedNotification>>() // Mediatr registers handlers as Transient by default, this extension ensures these aren't still there, only needed when service != Transient
                    .Add<SqlQueueClient>()
                    .Singleton()
                    .AsSelf() // this is needed to create the instance the delegates resolve
                    .AsImplementedInterfaces();
```

Reply:
Good catch, thanks! Updated!